From 8213aaed4429ffae8ed01579e68f9d04ff2d2969 Mon Sep 17 00:00:00 2001
From: Jiri Popelka <jpopelka@redhat.com>
Date: Mon, 10 Oct 2016 17:36:15 +0200
Subject: [PATCH] Don't modify original requestHeaders in Requester.__log()

It's a problem when we need to use the requestHeaders later.

Like in case of __requestRaw() where we send a request + get response,
then __log() it and if the response contains status 301 (redirected)
send another request (with same requestHeaders) to redirected url.

If the __log() erases the 'Authorization' field, the subsequent
request fails.

Fixes #470
---
 github/Requester.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/github/Requester.py b/github/Requester.py
index 9fd77b1..1526f1c 100644
--- a/github/Requester.py
+++ b/github/Requester.py
@@ -352,6 +352,8 @@ class Requester:
         logger = logging.getLogger(__name__)
         if logger.isEnabledFor(logging.DEBUG):
             if "Authorization" in requestHeaders:
+                # don't modify original requestHeaders
+                requestHeaders = requestHeaders.copy()
                 if requestHeaders["Authorization"].startswith("Basic"):
                     requestHeaders["Authorization"] = "Basic (login and password removed)"
                 elif requestHeaders["Authorization"].startswith("token"):
-- 
2.9.3

